### ISSUE TITLE
FirebaseInitProvider not on main dex by default

### ISSUE THREAD

**Issue by softlion** [xamarin/xamarin-android#3188]
### Steps to Reproduce

1. Add firebase nuget, or add facebook login nuget (same problem)
2. Use google login or facebook login
3. Enable multidex, add enough classes to go past the 65K limitation (add android support libs for ex)
4. Enable D8+R8
5. Build and run

Note: you need to delete bin/obj folders after every change to proguard.cfg or multidex.keep files. For some reason, xamarin tooling does not detect/use the changes.

### Expected Behavior
Runs fine

### Actual Behavior
Fails to run with "not on main dex list" error for both 
com/google/firebase/provider/FirebaseInitProvider.class
and
com/facebook/internal/FacebookInitProvider.class

### Version Information
VS2019 16.1.2

### Log File

06-08 19:56:34.762 18387 18387 E AndroidRuntime: java.lang.RuntimeException: Unable to get provider com.google.firebase.provider.FirebaseInitProvider: java.lang.ClassNotFoundException: Didn't find class "com.google.firebase.provider.FirebaseInitProvider" on path: DexPathList[[zip file "/data/app/com.tracks.recorder-1/base.apk"],nativeLibraryDirectories=[/data/app/com.tracks.recorder-1/lib/x86_64, /data/app/com.tracks.recorder-1/base.apk!/lib/x86_64, /system/lib64, /vendor/lib64]]

### Workaround
Create a multidex.keep file containing the existing classes to keep in maindexlist, which can be obtained by opening the "obj" folder, the add the above classes.

The problem that Xamarin tooling have there, is that is should be able to MERGE the generated maindexlist with the user's maindexlist.

**Comment by softlion** [comment]
So, 
it seem the issue is more the agressive optimizations caused by R8, which is more what we are used to with iOS.
But because R8 is new, the legacy support libs and other SDKs are not "R8 aware" and lots of methods / properties are optimized away.

So the best is to disable R8 and use proguard instead.

**Comment by devology-rob** [comment]
I'm experiencing this issue, my settings are...

Generate on package (.pak) per selected ABI
Enable Multi-Dex
Dex Compiler d8
Code Shrinker r8

Link SDK assemblies only

In the latest Visual Studio for Mac I no longer see the ProGuard options and may need to resort to hacking the project file?

I don't think this was a problem until I updated all my Nugets, so I might roll back the Google Play libraries to see if that fixes things.  

Note this was not apparent in DEBUG mode, but I noticed it once deployed to Google Play ALPHA.  A retrospective 'RELEASE' build direct to the device shows the same issue.

Any more details on where the multidex.keep file needs to be place, whether it has to be set as any specific build type and whether it's as simple as adding just com.google.firebase.provider.FirebaseInitProvider or whether you copy over a ton of classes spewed out in the obj folder will be useful.
